eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/CleanUpRelationTypeEqZeroCommand.php

```diff
@@ -0,0 +1,160 @@
+
+<warning>During the script execution the database should not be modified.
+
+To avoid surprises you are advised to create a backup or execute a dry run:
+ 
+    %command.name% dry-run
+    
+before proceeding with actual update.</warning>
+
+Since this script can potentially run for a very long time, to avoid memory
+exhaustion run it in production environment using <info>--env=prod</info> switch.
+EOT
+            );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $action = $input->getArgument('action');
+
+        if (!$this->isValidAction($action)) {
+            throw new RuntimeException(
+                "Action '{$action}' is not supported, use one of: " .
+                implode(', ', $this->getAvailableActions())
+            );
+        }
+
+        $totalCount = $this->getTotalCount();
+        $output->writeln('Found total relations to delete: ' . $totalCount);
+        if ($totalCount == 0) {
+            $output->writeln('Nothing to process, exiting.');
+
+            return;
+        }
+
+        if ($this->confirmExecution($input, $output)) {
+            switch ($action) {
+                case 'fix':
+                    $this->executeFix();
+                    break;
+                case 'list':
+                case 'dry-run':
+                    $this->executeList($output, $totalCount);
+                    break;
+            }
+        }
+
+        $output->writeln('');
+    }
+
+    protected function confirmExecution(InputInterface $input, OutputInterface $output)
+    {
+        $question = new ConfirmationQuestion('<question>Are you sure you want to proceed?</question> ', false);
+
+        return $this
+            ->getHelper('question')
+            ->ask($input, $output, $question);
+    }
+
+    protected function executeFix()
+    {
+        /** @var \eZ\Publish\Core\Persistence\Database\DatabaseHandler $conn */
+        $conn = $this->getContainer()->get('ezpublish.connection');
+        $conn->exec('DELETE FROM ezcontentobject_link WHERE relation_type = 0');
+    }
+
+    protected function executeList(OutputInterface $output, $totalCount)
+    {
+        $output->writeln('Following relations will be removed: ');
+        $output->writeln("ID\tFCO_ID\tFCO_V\tTCO_ID\tRT");
+
+        $passSize = 1000;
+        $passCount = ceil($totalCount / $passSize);
+        for ($pass = 0; $pass <= $passCount; ++$pass) {
+            $relations = $this->getCorruptedRelations($pass * $passSize, $passSize);
+            foreach ($relations as $relation) {
+                $output->writeln(vsprintf("%s\t%s\t%s\t%s\t%s", $relation));
+            }
+            $relations = null;
+        }
+    }
+
+    protected function getCorruptedRelations($offset, $limit)
+    {
+        /** @var \eZ\Publish\Core\Persistence\Database\DatabaseHandler $connection */
+        $connection = $this->getContainer()->get('ezpublish.connection');
```

Comment:
Is it ? We don't check anywhere if the legacy storage engine is used. We don't have another one at the moment, but it will fail in some future.